### PR TITLE
Allow absent default routers

### LINKED ISSUE #239: Gateway shouldn't crash when router channel isn't defined
For router update channels like the newly added `testnet` channel, when the `router.testnet` uri and pubkey aren't defined in any settings config (default.toml or settings.toml, etc) the gateway should not immediately crash when it can't find the components to construct a KeyedUri for the default router on that chain.

The current behavior is that the process will immediately terminate with an error that `uri` is missing and will behave similarly if a uri is defined but a pubkey is not.

### PR DESCRIPTION
**Pull request by madninja**
Fixes #239

### CODE REVIEW

File: src/router/dispatcher.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ pub struct Dispatcher {
     gateways: Vec<KeyedUri>,
     routing_height: u64,
     region_height: u64,
-    default_router: KeyedUri,
+    default_router: Option<KeyedUri>,
```
Comment (JayKickliter) [inline src/router/dispatcher.rs:91]: OCD: I prefer seeing Option fields at the bottom of a struct

Comment (madninja) [reply src/router/dispatcher.rs:91]: OCD pacified!

File: src/settings.rs [hunk 2]
```diff
@@ -111,8 +111,10 @@ impl Settings {
             .map_err(|e| e.into())
     }
 
-    pub fn default_router(&self) -> &KeyedUri {
-        &self.router[&self.update.channel.to_string()]
+    pub fn default_router(&self) -> Option<KeyedUri> {
+        self.router
+            .get(&self.update.channel.to_string())
+            .map(|uri| uri.to_owned())
```
Comment (jeffgrunewald) [inline src/settings.rs:117]: this seems like a misnomer since it's not just going to get the uri element of the router keyeduri

Comment (madninja) [reply src/settings.rs:117]: I've interchangeably used `uri` before to indicate `keyed_uri`. Can't disagree it's a bit ambiguous. We'll have to take a pass at clearing this up.

Comment (madninja) [reply src/settings.rs:117]: and fixed in 1da305351aa112d15c788078ccea559235500813

**Pull request merged**